SIPS/sip-63.md

```diff
@@ -0,0 +1,52 @@
+---
+sip: 63
+title: Trading Incentives
+status: WIP
+author: Kain Warwick (@kaiynne)
+discussions-to: TBC
+
+created: 2020-05-28
+---
+
+## Simple Summary
+
+This SIP proposes a change to the Feepool contract to record the fees paid by each address to enable the distribution of trading incentives.
+
+## Abstract
+The feepool contract will be updated to write to a Trading Incentives contract after each succesful exchange. Feepool will read the current trading fees paid in that period by the address and add the trading fees paid in that exchange. The trading incentives contract will be a modified version of the LPRewards contract, each time fees are paid the balance of fees paid will be added to the Trading Incentives contract. SNX will bde deposited manually each fee period into this contract and traders will earn a portion of this SNX as they pay trading fees. 
+
+## Motivation
+There is currently no way to track trading fees paid on-chain, so it is not possible to reward traders who pay more fees on the exchange. Paying fee rebates is a very powerful way of reducing friction and switching costs for new traders. Hwoever, where CEX's can identify users and pay onboarding fees and other incentives a DEX does not have this ability so using a pooled mechanism rather than a direct rebate per user ensures the total fee incentive does not exceed a specified amount and cannot be sybil attacked. The pooled mechanism also ensures that traders who trade earlier are rewarded more than later traders incentivising traders to test out the exchange sooner. While the sX trading experience has improved significantly, there is still friction due to latency and other limitations. While we fully expect these limitations to be reduced or eliminated entirely in the near future, attracting new traders is critical to the growth of the platform. The trading experience is now sufficiently differentiated that the experience is likely to be positive relative to other DEX's particularly with respect to slippage and fees. This pooled mechanism also introduce the ability to add referal incentives and other mechanism if the base incentive proves successful. 
+
+## Specification
+<!--The specification should describe the syntax and semantics of any new feature, there are five sections
+1. Overview
+2. Rationale
+3. Technical Specification
+4. Test Cases
+5. Configurable Values
+-->
+
+### Overview
+Feepool will be modified to write to a Trading Incentive contract each time a trade is completed. The function will check the total fees paid in that fee period and add the additional fees paid in that exchange to this balance.
+
+This fee balance will be used to calculate the distribution of the SNX during that period. At the end of the period the balance will be wiped for all addresses and fees will begin to accumulate again. Balances will be zeroed when the new SNX is deposited from the inflationary supply or manually during the trial period.
```

Comment:
Just a nitpick, but I highly doubt we will be wiping or zeroing individual address mappings - it would be too cumbersome - we'd just have a flag to indicate a change and perform a different calculation in the function. I understand what you've written is more conceptual, but we do need to be cognizant of what changes are being proposed on-chain, as some in the community may take it literally.

Reply:
Fixed, I think @jacko125 has a view on how to improve this so will keep it more generic.